gtsfm/averaging/translation/averaging_1dsfm.py

```diff
@@ -171,46 +166,272 @@ def _get_prior_measurements_in_world_frame(
 
-    def __get_initial_values(self, wTi_initial: List[Optional[PosePrior]]):
+        # Compute outlier weights using MFAS.
+        # TODO(ayush): parallelize this step.
+        outlier_weights: List[Dict[Tuple[int, int], float]] = []
+        for direction in projection_directions:
+            mfas_instance = MFAS(w_i1Ui2_measurements, direction)
+            outlier_weights.append(mfas_instance.computeOutlierWeights())
+        logger.debug("Computed outlier weights using MFAS.")
+
+        # Compute average outlier weight.
+        outlier_weights_sum: DefaultDict[Tuple[int, int], float] = defaultdict(float)
+        inliers = set()
+        for outlier_weight_dict in outlier_weights:
+            for w_i1Ui2 in w_i1Ui2_measurements:
+                i1, i2 = w_i1Ui2.key1(), w_i1Ui2.key2()
+                outlier_weights_sum[(i1, i2)] += outlier_weight_dict[(i1, i2)]
+        for (i1, i2), weight_sum in outlier_weights_sum.items():
+            if weight_sum / len(projection_directions) < OUTLIER_WEIGHT_THRESHOLD:
+                inliers.add((i1, i2))
+
+        # Filter outliers, index back from symbol to int.
+        # `inliers` contains both camera-camera and camera-landmark inliers. We separate them here.
+        inlier_w_i2Ui1_dict = {}
+        inlier_w_i2Ui1_dict_tracks = {}
+        inlier_cameras: Set[int] = set()
+        for (i1, i2) in w_i2Ui1_dict:
+            if (C(i2), C(i1)) in inliers:  # there is a flip in indices from w_i2Ui1_dict to inliers.
+                inlier_w_i2Ui1_dict[(i1, i2)] = w_i2Ui1_dict[(i1, i2)]
+                inlier_cameras.add(i1)
+                inlier_cameras.add(i2)
+
+        for (track_id, cam_id) in w_i2Ui1_dict_tracks:
+            # Same as above, `inliers` contains symbols that are flipped - C(cam_id), L(track_id).
+            # Only add an inlier camera-track measurements if the camera has other camera-camera inliers.
+            if (C(cam_id), L(track_id)) in inliers and cam_id in inlier_cameras:
+                inlier_w_i2Ui1_dict_tracks[(track_id, cam_id)] = w_i2Ui1_dict_tracks[(track_id, cam_id)]
+
+        return inlier_w_i2Ui1_dict, inlier_w_i2Ui1_dict_tracks, inlier_cameras
+
+    def __get_initial_values(self, wTi_initial: List[Optional[PosePrior]]) -> gtsam.Values:
         """Converts translations from a list of absolute poses to gtsam.Values for initialization.
 
         Args:
-            wTi_initial: List of
+            wTi_initial: List of absolute poses.
+
+        Returns:
+            gtsam.Values containing initial translations (uses symbols for camera index).
         """
         initial = gtsam.Values()
         for i, wTi in enumerate(wTi_initial):
             if wTi is not None:
-                initial.insertPoint3(i, wTi.value.translation())
+                initial.insertPoint3(C(i), wTi.value.translation())
         return initial
 
+    def _select_tracks_for_averaging_by_measurements(
```

Comment:
It is not clear to me why we use the prefix `_by_measurements`

Reply:
because I had another function named _select_tracks_for_averaging before this. removed it now.